src/main/java/org/jenkinsci/plugins/workflow/flow/StashManager.java

```diff
@@ -109,26 +114,36 @@ public static void stash(@Nonnull Run<?,?> build, @Nonnull String name, @Nonnull
      * @see StashAwareArtifactManager#stash
      */
     @SuppressFBWarnings(value="RV_RETURN_VALUE_IGNORED_BAD_PRACTICE", justification="fine if mkdirs returns false")
-    public static void stash(@Nonnull Run<?,?> build, @Nonnull String name, @Nonnull FilePath workspace, @Nonnull Launcher launcher, @Nonnull EnvVars env, @Nonnull TaskListener listener,
+    public static List<File> stash(@Nonnull Run<?,?> build, @Nonnull String name, @Nonnull FilePath workspace, @Nonnull Launcher launcher, @Nonnull EnvVars env, @Nonnull TaskListener listener,
                              @CheckForNull String includes, @CheckForNull String excludes, boolean useDefaultExcludes, boolean allowEmpty) throws IOException, InterruptedException {
         Jenkins.checkGoodName(name);
+        List<File> files = new ArrayList<>();
+        DirScanner.Glob scanner = new DirScanner.Glob(Util.fixEmpty(includes) == null ? "**" : includes, excludes, useDefaultExcludes);
+        scanner.scan(new File(workspace.getRemote()), new FileVisitor() {
```

Comment:
`new File(workspace.getRemote())` is a sure sign that something is wrong. This takes a file path on an agent (in general) and treats it as a local file. Which will result, in, say, `C:\home\jenkins\something` or `/var/jenkins_home/C:/Program Files/something` or similar junk.

Reply:
this is going away, all that will be done in Jenkins Core, where it actually happens.

Reply:
Saw that, and the same problem applies there. Do not use `File`.